protocol/rpcprovider/rpcprovider.go

```diff
@@ -165,6 +165,16 @@ func (rpcp *RPCProvider) Start(ctx context.Context, txFactory tx.Factory, client
 			chainCommonSetup := func() error {
 				chainMutexes[chainID].Lock()
 				defer chainMutexes[chainID].Unlock()
+
+				// Create temp chain fetcher so we can validate chainID
+				tempChainFetcher := chainlib.NewChainFetcher(ctx, chainRouter, chainParser, rpcProviderEndpoint)
```

Comment:
we are creating the same chainFetcher twice on the first apiInterface both here and on 189, we can use the one created here instead when storing
so no need to call it temp as we save one of them